core/logic/src/main/java/org/apache/syncope/core/logic/MailTemplateLogic.java

```diff
@@ -83,6 +84,13 @@ public MailTemplateTO transform(final MailTemplate input) {
 
     @PreAuthorize("hasRole('" + StandardEntitlement.MAIL_TEMPLATE_CREATE + "')")
     public MailTemplateTO create(final String key) {
+        for (MailTemplateTO mailTemplateTO: list()) {
```

Comment:
Take a look at [ResourceLogic](https://github.com/apache/syncope/blob/master/core/logic/src/main/java/org/apache/syncope/core/logic/ResourceLogic.java#L121-L123) and do accodingly.